deployment/common/changeset/transfer_link_token.go

```diff
@@ -0,0 +1,71 @@
+package changeset
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+var _ deployment.ChangeSet[TransferLinkTokenBatchConfig] = TransferLinkTokenBatch
+
+type TokenTransferer interface {
+	Transfer(opts *bind.TransactOpts, to common.Address, amount *big.Int) (*gethtypes.Transaction, error)
+}
+
+type TransferLinkTokenConfig struct {
+	To         common.Address
+	Amount     *big.Int
+	Transferer TokenTransferer
+}
+
+type TransferLinkTokenBatchConfig struct {
+	Transfers map[uint64]TransferLinkTokenConfig
+}
+
+func (cfg TransferLinkTokenBatchConfig) Validate() error {
+	for k, c := range cfg.Transfers {
+		if err := deployment.IsValidChainSelector(k); err != nil {
+			return err
+		}
+		if c.To == (common.Address{}) {
+			return errors.New("to address must be set")
+		}
+		if c.Amount == nil || c.Amount.Sign() == -1 {
+			return errors.New("amount must be set and positive")
+		}
+		if c.Transferer == nil {
+			return errors.New("transferer must be set")
+		}
+	}
+
+	return nil
+}
+
+// TransferLinkTokenBatch transfers link token to the to address from the transferer in a batch.
+func TransferLinkTokenBatch(e deployment.Environment, config TransferLinkTokenBatchConfig) (deployment.ChangesetOutput, error) {
+	if err := config.Validate(); err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid transfer link token config: %v", err)
+	}
+
+	for chainSelector, transferConfig := range config.Transfers {
+		chain, ok := e.Chains[chainSelector]
+		if !ok {
+			return deployment.ChangesetOutput{}, fmt.Errorf("chain not found in environment")
+		}
+		tx, err := transferConfig.Transferer.Transfer(chain.DeployerKey, transferConfig.To, transferConfig.Amount)
```

Comment:
Is any information from the `tx` outputted somewhere? Maybe logs, result? If not, we probably should to help us debugging, and not only in case of error

Reply:
The `tx` and `err` are passed into `deployment.ConfirmIfNoError` and if there is any issue, msg will be outputted.

eg 
```
        	Error:      	Received unexpected error:
        	            	failed to transfer link token to 0x0000000000000000000000000000000000000001: execution reverted: ERC20: transfer amount exceeds balance
        	Test:       	TestTransferLinkToken
```

Reply:
so i follow the other pattern in the repo, where i pass in the tx and error into `deployment.ConfirmIfNoError(chain, tx, err)`  and if there is a issue, it will return an error with description in there, eg 

```
failed to confirm transfer link token to 0x0000000000000000000000000000000000000001: transaction reverted: Error execution reverted: ERC20: transfer amount exceeds balance ErrorData 0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002645524332303a207472616e7366657220616d6f756e7420657863656564732062616c616e63650000000000000000000000000000000000000000000000000000
```

Reply:
Ideally, at least tx hashes are always outputted. Not only in case of errors, but on successes, can be useful for many things, from debugging to tx event searching. I think this is something should be done at the framework level, each changeset implementation shouldn't have its own way of managing this.

In any case, for this one, let's make sure we output the tx hash in case of error.